sample-app/src/main.js

```diff
@@ -83,12 +83,15 @@ const questions1 = [
   },
   {
       type: "input",
+      guiOptions: {
+        applyDefault: true
```

Comment:
We apply the default value/function when the form is loaded, and even while the user changes the value of another question. However, we do not do this when the question is dirty.

The meaning of the gui option you introduced is that we apply the default value when the question _is_ dirty; therefore, I propose that you rename the option to `applyDefaultWhenDirty`.